pkg/types/binary_test.go

```diff
@@ -0,0 +1,27 @@
+package types
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestBinary_MarshalJSON(t *testing.T) {
+	subtests := []struct {
+		input  Binary
+		output string
+	}{
+		{nil, `null`},
+		{make(Binary, 0, 1), `null`},
+		{Binary(" "), `"20"`},
+	}
+
+	for _, st := range subtests {
+		t.Run(fmt.Sprintf("Binary(%#v)", []byte(st.input)), func(t *testing.T) {
+			actual, err := st.input.MarshalJSON()
+
+			require.NoError(t, err)
+			require.Equal(t, []byte(st.output), actual)
```

Comment:
Casting towards `string` instead of `[]byte` results in more readable output when a test fails, compare:

#### []byte
```
    --- FAIL: TestBinary_MarshalJSON/Binary([]byte(nil)) (0.00s)
        binary_test.go:24: 
            	Error Trace:	/home/jbrost/src/icingadb/pkg/types/binary_test.go:24
            	Error:      	Not equal: 
            	            	expected: []byte{0x6e, 0x75, 0x6c, 0x6c}
            	            	actual  : []byte{0x6e, 0x69, 0x6c}
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1,3 +1,3 @@
            	            	-([]uint8) (len=4) {
            	            	- 00000000  6e 75 6c 6c                                       |null|
            	            	+([]uint8) (len=3) {
            	            	+ 00000000  6e 69 6c                                          |nil|
            	            	 }
            	Test:       	TestBinary_MarshalJSON/Binary([]byte(nil))
```

#### string
```
    --- FAIL: TestBinary_MarshalJSON/Binary([]byte(nil)) (0.00s)
        binary_test.go:24: 
            	Error Trace:	/home/jbrost/src/icingadb/pkg/types/binary_test.go:24
            	Error:      	Not equal: 
            	            	expected: "null"
            	            	actual  : "nil"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-null
            	            	+nil
            	Test:       	TestBinary_MarshalJSON/Binary([]byte(nil))
```